File: android/app/src/main/java/com/bensbible/app/model/ShareGradient.kt
package com.bensbible.app.model

import androidx.compose.ui.graphics.Color

enum class ShareGradient {
    NAVY,
    SUNSET,
    FOREST,
    PARCHMENT,
    MIDNIGHT;

    val colors: List<Color>
        get() = when (this) {
            NAVY -> listOf(Color(0.05f, 0.1f, 0.3f), Color(0.15f, 0.25f, 0.55f))
            SUNSET -> listOf(Color(0.95f, 0.5f, 0.2f), Color(0.85f, 0.25f, 0.4f))
            FOREST -> listOf(Color(0.05f, 0.2f, 0.15f), Color(0.1f, 0.4f, 0.35f))
            PARCHMENT -> listOf(Color(0.96f, 0.93f, 0.85f), Color(0.88f, 0.82f, 0.7f))
            MIDNIGHT -> listOf(Color(0.05f, 0.02f, 0.1f), Color(0.2f, 0.08f, 0.35f))
        }

    val textColor: Color
        get() = when (this) {
            PARCHMENT -> Color(0.2f, 0.15f, 0.1f)
            else -> Color.White
        }

    val accentColor: Color
        get() = when (this) {
            NAVY -> Color(0.6f, 0.75f, 1.0f)
            SUNSET -> Color(1.0f, 0.9f, 0.7f)
            FOREST -> Color(0.5f, 0.85f, 0.7f)
            PARCHMENT -> Color(0.55f, 0.4f, 0.25f)
            MIDNIGHT -> Color(0.7f, 0.55f, 0.95f)
        }
}

File: BensBible2/Models/ShareGradient.swift
import SwiftUI

enum ShareGradient: String, CaseIterable, Identifiable {
    case navy
    case sunset
    case forest
    case parchment
    case midnight

    var id: String { rawValue }

    var colors: [Color] {
        switch self {
        case .navy: [Color(red: 0.05, green: 0.1, blue: 0.3), Color(red: 0.15, green: 0.25, blue: 0.55)]
        case .sunset: [Color(red: 0.95, green: 0.5, blue: 0.2), Color(red: 0.85, green: 0.25, blue: 0.4)]
        case .forest: [Color(red: 0.05, green: 0.2, blue: 0.15), Color(red: 0.1, green: 0.4, blue: 0.35)]
        case .parchment: [Color(red: 0.96, green: 0.93, blue: 0.85), Color(red: 0.88, green: 0.82, blue: 0.7)]
        case .midnight: [Color(red: 0.05, green: 0.02, blue: 0.1), Color(red: 0.2, green: 0.08, blue: 0.35)]
        }
    }

    var textColor: Color {
        switch self {
        case .parchment: Color(red: 0.2, green: 0.15, blue: 0.1)
        default: .white
        }
    }

    var accentColor: Color {
        switch self {
        case .navy: Color(red: 0.6, green: 0.75, blue: 1.0)
        case .sunset: Color(red: 1.0, green: 0.9, blue: 0.7)
        case .forest: Color(red: 0.5, green: 0.85, blue: 0.7)
        case .parchment: Color(red: 0.55, green: 0.4, blue: 0.25)
        case .midnight: Color(red: 0.7, green: 0.55, blue: 0.95)
        }
    }
}
